src/panels/config/logs/error-log-card.ts

```diff
@@ -176,13 +186,32 @@ class ErrorLogCard extends LitElement {
                     </ha-menu>
                   `
                 : nothing}
-              <ha-icon-button
-                .path=${mdiDownload}
-                @click=${this._downloadFullLog}
-                .label=${this.hass.localize(
-                  "ui.panel.config.logs.download_full_log"
-                )}
-              ></ha-icon-button>
+              ${this._downloadSupported
+                ? html`
+                    <ha-icon-button
+                      .path=${mdiDownload}
+                      @click=${this._downloadLogs}
+                      .label=${this.hass.localize(
+                        "ui.panel.config.logs.download_logs"
+                      )}
+                    ></ha-icon-button>
+                  `
+                : this._logsFileLink
+                  ? html`
+                      <a
+                        href=${this._logsFileLink}
+                        target="_blank"
+                        class="download-link"
```

Comment:
Why can't we do this in the `fileDownload` helper function? That is also just a anchor link? Is it because it is pressed by JS and not by a user?

Reply:
Yes. It is only possible when you actually click it. I tried multiple ways and this was the only that worked. Otherwise the dialog would be an option.